pyomo/contrib/gdpopt/nlp_initialization.py

```diff
@@ -0,0 +1,34 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+"""A library of possible callbacks to use for initializing the NLP subproblems.
+However, it seems likely that problem-specific routines might be better, in
+which case you can write your own, and specify it in the
+'subproblem_initialization_method' argument."""
+
+# This is the original GDPopt behavior:
+def restore_vars_to_original_values(nlp_util_block, mip_util_block):
```

Comment:
May I suggest another name besides `original`? I would go use `initial` instead. It just sounds odd to me

Reply:
I'm not sure I like using `initial` because we could set the values to something other than what was in the original (given) model and we would still call whatever we set them to the "initial value." Would `given` or `user_given` sound better to you?